core-bundle/tests/Twig/Extension/ContaoExtensionTest.php

```diff
@@ -377,25 +377,33 @@ public static function provideTemplateNames(): iterable
      *
      * @dataProvider provideTwigFunctionSignatures
      */
-    public function testContaoUsesCorrectTwigFunctionSignatures(string $function, array $expectedParameters): void
+    public function testContaoUsesCorrectTwigFunctionSignatures(\ReflectionFunction|\ReflectionMethod $reflector, array $expectedParameters): void
     {
-        // Make sure the functions outside the class scope are loaded
-        new \ReflectionClass(EscaperExtension::class);
-
         $parameters = array_map(
             static fn (\ReflectionParameter $parameter): array => [
                 ($type = $parameter->getType()) instanceof \ReflectionNamedType ? $type->getName() : null,
                 $parameter->getName(),
             ],
-            (new \ReflectionFunction($function))->getParameters(),
+            $reflector->getParameters(),
         );
+
         $this->assertSame($parameters, $expectedParameters);
     }
 
     public static function provideTwigFunctionSignatures(): iterable
     {
+        // Make sure the functions outside the class scope are loaded
+        new \ReflectionClass(EscaperExtension::class);
+
+        // Forward compatibility with twig/twig 4
+        if (method_exists(EscaperExtension::class, 'escape')) {
+            $escape = new \ReflectionMethod(EscaperExtension::class.'::escape');
```

Comment:
This usage seems to be deprecated in PHP 8.4.

```suggestion
            $escape = new \ReflectionMethod(EscaperExtension::class, 'escape');
```